src/restore.ts

```diff
@@ -12,6 +13,16 @@ process.on("uncaughtException", (e) => {
 });
 
 async function run() {
+  const cache = function () {
```

Comment:
Moving this out into a `getCacheBackend()` function that is used by both restore/save.
If can also add the logging part to the current place where all the informational logging happens.